ios/sound.mm

```diff
@@ -24,209 +24,277 @@
 {
-  /*// This function checks if our required input/output channel
-  // properties are supported by the selected device. If the return
-  // string is empty, the device can be used, otherwise the error
-  // message is returned.
-
-  // check sample rate of input
-  audioSession.setPreferredSampleRate( SYSTEM_SAMPLE_RATE_HZ );
-  // if false, try to set it
-  if ( audioSession.sampleRate != SYSTEM_SAMPLE_RATE_HZ )
-  {
-    throw CGenErr ( tr ("Could not set sample rate. Please close other applications playing audio.") );
-  }
-
-  // special case with 4 input channels: support adding channels
-    if ( ( lNumInChan == 4 ) && bInputChMixingSupported )
+    try
     {
-        // add four mixed channels (i.e. 4 normal, 4 mixed channels)
-        lNumInChanPlusAddChan = 8;
-
-        for ( int iCh = 0; iCh < lNumInChanPlusAddChan; iCh++ )
-        {
-            int iSelCH, iSelAddCH;
-
-            GetSelCHAndAddCH ( iCh, lNumInChan, iSelCH, iSelAddCH );
-
-            if ( iSelAddCH >= 0 )
-            {
-                // for mixed channels, show both audio channel names to be mixed
-                channelInputName[iCh] =
-                    channelInputName[iSelCH] + " + " + channelInputName[iSelAddCH];
-            }
-        }
+        NSError* audioSessionError = nil;
+
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
+        [[AVAudioSession sharedInstance] requestRecordPermission:^( BOOL granted ) {
+          if ( granted )
+          {
+              // ok
+          }
+        }];
+        [[AVAudioSession sharedInstance] setMode:AVAudioSessionModeMeasurement error:&audioSessionError];
     }
-    else
+    catch ( const CGenErr& generr )
     {
-        // regular case: no mixing input channels used
-        lNumInChanPlusAddChan = lNumInChan;
+        qDebug ( "Sound exception ...." );
     }
-*/
-    // everything is ok, return empty string for "no error" case
-    return "";
+    isInitialized = false;
 }
 
-void CSound::SetLeftInputChannel  ( const int iNewChan )
+int CSound::Init ( const int iCoreAudioBufferSizeMono )
 {
-    /*// apply parameter after input parameter check
-    if ( ( iNewChan >= 0 ) && ( iNewChan < iNumInChanPlusAddChan ) )
+    try
     {
-        iSelInputLeftChannel = iNewChan;
+        // printf("Init sound ..."); <= to check the number of Sound inits at launch
+        // init base class
+        // CSoundBase::Init ( iCoreAudioBufferSizeMono ); this does nothing
+        this->iCoreAudioBufferSizeMono = iCoreAudioBufferSizeMono;
+
+        // set internal buffer size value and calculate stereo buffer size
+        iCoreAudioBufferSizeStereo = 2 * iCoreAudioBufferSizeMono;
+
+        // create memory for intermediate audio buffer
+        vecsTmpAudioSndCrdStereo.Init ( iCoreAudioBufferSizeStereo );
+
+        AVAudioSession* sessionInstance = [AVAudioSession sharedInstance];
```

Comment:
See, I'm happy with this spacing.  It's equivalent of `something = other.methodWithReturnValue();`
...